src/steth/scDAI.sol

```diff
@@ -0,0 +1,564 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.19;
+
+import {NoProfitsToSell, FlashLoanAmountZero, EndDaiBalanceTooLow, FloatBalanceTooLow} from "../errors/scErrors.sol";
+
+import {ERC20} from "solmate/tokens/ERC20.sol";
+import {WETH} from "solmate/tokens/WETH.sol";
+import {ERC4626} from "solmate/mixins/ERC4626.sol";
+import {SafeTransferLib} from "solmate/utils/SafeTransferLib.sol";
+import {FixedPointMathLib} from "solmate/utils/FixedPointMathLib.sol";
+import {Address} from "openzeppelin-contracts/utils/Address.sol";
+import {EnumerableMap} from "openzeppelin-contracts/utils/structs/EnumerableMap.sol";
+
+import {Constants as C} from "../lib/Constants.sol";
+import {BaseV2Vault} from "./BaseV2Vault.sol";
+import {AggregatorV3Interface} from "../interfaces/chainlink/AggregatorV3Interface.sol";
+import {IAdapter} from "./IAdapter.sol";
+import {PriceConverter} from "./PriceConverter.sol";
+import {Swapper} from "./Swapper.sol";
+
+/**
+ * @title Sandclock Dai Vault
+ * @dev code mostly copied from scUSDCv2
+ * @notice deposit tokens -> DAI & sDAI
+ * @dev ISSUES: We are not using Lifi for swapping profits during withdrawals (beforeWithdraw) & exitAllPositions
+ * @dev in case of loss higher than 5% there is not enough weth after flashloan to payback flashloan
+ */
+contract scDAI is BaseV2Vault {
```

Comment:
scUSDC vault is already deployed and running well. Consider creating a base contract that will contain common functionalities for scDAI & scUSDC as well as any other vaults we may deploy in the future that will rely on scETH for generating yield

Reply:
Can we do this later as a separate pr?

Reply:
ideally, we would do it now but I'm fine with doing it later in a separate PR